lib/libspl/include/umem.h

```diff
@@ -196,7 +202,10 @@ umem_cache_free(umem_cache_t *cp, void *ptr)
 	if (cp->cache_destructor)
 		cp->cache_destructor(ptr, cp->cache_private);
 
-	umem_free(ptr, cp->cache_bufsize);
+	if (cp->cache_align != 0)
```

Comment:
Theoretically, we could request an alignment of 0 with the aligned function, which would make this pick the wrong free function, but I imagine that we do not use the API that way.